gitbutler-ui/src/lib/utils/context.ts

```diff
@@ -9,10 +20,43 @@ export function getContextByClass<T extends new (...args: any) => InstanceType<T
 	return instance;
 }
 
-export function getContextStoreByClass<T extends new (...args: any) => InstanceType<T>>(
-	key: T
-): Readable<InstanceType<T>> {
-	const instance = svelteGetContext<Readable<InstanceType<T>> | undefined>(key);
+/**
+ * Getter that returns an readable store of the parameter type
+ */
+// export function getContextStore<T extends new (...args: any) => InstanceType<T>>(
+// 	key: T
+// ): Readable<InstanceType<T>> {
+// 	const instance = svelteGetContext<Readable<InstanceType<T>> | undefined>(key);
+// 	if (!instance) throw new Error(`no instance of \`Readable<${key.name}>\` in context`);
+// 	return instance;
+// }
+
+export function getContextStore<
+	T extends new (...args: any) => InstanceType<T>,
+	S extends Readable<InstanceType<T>> = Readable<InstanceType<T>>
+>(key: T): S {
```

Comment:
If we're going for the name `getContextStore` over `getContextStoreByClass`, it feels appropriate that we should also rename `getContextByClass` to just `getContext`.

Reply:
Done!